packages/api-graphql/__tests__/internals/generateClient.test.ts

```diff
@@ -835,6 +834,186 @@ describe('generateClient', () => {
 		});
 	});
 
+	describe('error handling', () => {
+		test('create() returns null with errors property', async () => {
+			const expectedErrors = [
+				{
+					path: ['createTodo'],
+					data: null,
+					errorType: 'Unauthorized',
+					errorInfo: null,
+					locations: [
+						{
+							line: 2,
+							column: 3,
+							sourceName: null,
+						},
+					],
+					message: 'Not Authorized to access createTodo on type Mutation',
+				},
+			];
+
+			const spy = mockApiResponse({
+				data: {
+					getTodo: null,
+				},
+				errors: expectedErrors,
+			});
+
+			const client = generateClient<Schema>({ amplify: Amplify });
+			const { data, errors } = await client.models.Todo.create({
+				id: 'does not matter',
+			});
+
+			expect(data).toBeNull();
+			expect(errors?.length).toBe(1);
+			expect(errors).toEqual(expectedErrors);
+		});
+
+		test('get() returns null with errors property', async () => {
```

Comment:
nit: this test description is not straightforward to understand what's the test without looking the test body... wdyt?

```suggestion
		test('returns object in which `data` is null and `errors` contains expected error', async () => {
```